xio-core/src/test/java/com/xjeffrose/xio/http/ReverseProxyFunctionalTest.java

```diff
@@ -272,105 +274,108 @@ public void testHttp1toHttp2ServerPostMany() throws Exception {
-                        if (post) {
-                          MediaType mediaType = MediaType.parse("text/plain");
-                          RequestBody body = RequestBody.create(mediaType, "this is the post body");
-                          request.post(body);
-                        } else {
-                          request.get();
-                        }
-                        Response response = client.newCall(request.build()).execute();
-                        responses.put(index, response);
-                        waiter.resume();
-                      } catch (Exception error) {
-                        waiter.fail(error);
-                      }
-                    }));
-
-    int seconds = 10;
-    Exception timeout = null;
-    try {
-      waiter.await(seconds, TimeUnit.SECONDS, NUM_REQUESTS);
-    } catch (Exception e) {
-      timeout = e;
-    }
+  private void verify(Iterable<Pair> responses) {
+    assertEquals(NUM_REQUESTS, Streams.stream(responses).count());
     responses.forEach(
-        (key, response) -> {
-          String index = response.header("x_index");
+        pair -> {
+          String index = pair.response.header("x_index");
           assertNotNull(index);
-          assertEquals(key.toString(), index);
+          assertEquals(pair.index.toString(), index);
         });
-    assertEquals("expected a response for all of the requests", NUM_REQUESTS, responses.size());
-    executorService.shutdown();
-    executorService.awaitTermination(seconds, TimeUnit.SECONDS);
-    if (timeout != null) {
-      log.error("timeout", timeout);
-      fail("test timed out");
+  }
+
+  private Observable<Pair> requests(boolean post) throws Exception {
+    String url = url(port(), false);
+    return Observable.fromIterable(() -> IntStream.range(0, NUM_REQUESTS).iterator())
+        .flatMapSingle(
+            index ->
+                Single.<Pair>create(
+                        emitter -> {
+                          Request.Builder request =
+                              new Request.Builder()
+                                  .header("x_index", String.valueOf(index))
+                                  .url(url);
+                          if (post) {
+                            MediaType mediaType = MediaType.parse("text/plain");
+                            RequestBody body =
+                                RequestBody.create(mediaType, "this is the post body");
+                            request.post(body);
+                          } else {
+                            request.get();
+                          }
+                          Response response = client.newCall(request.build()).execute();
+                          log.debug("response {}", response);
+                          emitter.onSuccess(new Pair(index, response));
+                        })
+                    .subscribeOn(Schedulers.io()));
```

Comment:
sultle nuance to rxjava: `subscribeOn` can only be applied once to the upstream source ... this schedules work on the rx "io" thread pool